tests/system_tests/test_functional/test_tensorboard/test_tf_summary.py

```diff
@@ -239,50 +231,54 @@ def test_compat_tensorboard(relay_server, wandb_init):
                     step,
                 )
 
-    run_ids = relay.context.get_run_ids()
-    assert len(run_ids) == 1
-    assert run.id == run_ids[0]
+    with wandb_backend_spy.freeze() as snapshot:
+        assert len(snapshot.run_ids()) == 1
 
-    summary = relay.context.get_run_summary(run.id)
-    assert summary["global_step"] == 9
-    assert "x_scalar_1" in summary
+        summary = snapshot.summary(run_id=run.id)
+        assert summary["global_step"] == 9
+        assert "x_scalar_1" in summary
 
-    history = relay.context.get_run_history(run.id)
-    assert len(history) == 10
+        history = snapshot.history(run_id=run.id)
+        assert len(history) == 10
 
     wandb.tensorboard.unpatch()
 
 
 def test_tb_sync_with_explicit_step_and_log(
     wandb_init,
-    relay_server,
+    wandb_backend_spy,
     mock_wandb_log,
 ):
-    with relay_server() as relay:
-        with wandb_init(sync_tensorboard=True) as run:
-            with tf.summary.create_file_writer(
-                "test/logs",
-            ).as_default():
-                for i in range(10):
-                    tf.summary.scalar(
-                        "x_scalar",
-                        tf.constant(i**2),
-                        step=i,
-                    )
-            run.log({"y_scalar": 1337}, step=42)
-
-    assert mock_wandb_log.warned("Step cannot be set when using tensorboard syncing")
-    history = relay.context.get_run_history(run.id)
-    assert len(history) == 11
-    assert history["x_scalar"].dropna().tolist() == [i**2 for i in range(10)]
-    assert history["y_scalar"].dropna().tolist() == [1337]
-
-    summary = relay.context.get_run_summary(run.id)
-    assert summary["global_step"] == 9
-    assert summary["x_scalar"] == 81
-    assert summary["y_scalar"] == 1337
-
-    telemetry = relay.context.get_run_telemetry(run.id)
-    assert 35 in telemetry["3"]  # sync_tensorboard
+    with wandb_init(sync_tensorboard=True) as run:
+        with tf.summary.create_file_writer(
+            "test/logs",
+        ).as_default():
+            for i in range(10):
+                tf.summary.scalar(
+                    "x_scalar",
+                    tf.constant(i**2),
+                    step=i,
+                )
+        run.log({"y_scalar": 1337}, step=42)
+
+    with wandb_backend_spy.freeze() as snapshot:
+        assert mock_wandb_log.warned(
+            "Step cannot be set when using tensorboard syncing"
+        )
+        history = snapshot.history(run_id=run.id)
+        assert len(history) == 11
+
+        history_df = pd.DataFrame.from_dict(history, orient="index")
+
+        assert history_df["x_scalar"].dropna().tolist() == [i**2 for i in range(10)]
+        assert history_df["y_scalar"].dropna().tolist() == [1337]
```

Comment:
Here, ideally we would do something like:

```python
assert history == {
    0: {"x_scalar": ..., "global_step": ...},
    1: ...
}
```

but unfortunately we can't predict the order in which `run.log()` happens relative to the TB event processing logic. Still, if you only call `tf.summary.scalar()` once, you could check that `history` is one of two possibilities:

```python
x_scalar_row = {"x_scalar": ..., "global_step": ...}
y_scalar_row = {"y_scalar": 1337}
assert history in (
    {0: x_scalar_row, 1: y_scalar_row},
    {0: y_scalar_row, 1: x_scalar_row},
)
```

(I think that syntax should work, but I haven't checked)

And I think you can drop the `summary` check; we don't emit summary records from the TB integration, and the codepath that computes summary values is thoroughly tested elsewhere.